pkg/auth/event_policy.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2024 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/eventing/pkg/apis/eventing/v1alpha1"
+	listerseventingv1alpha1 "knative.dev/eventing/pkg/client/listers/eventing/v1alpha1"
+	"knative.dev/pkg/resolver"
+)
+
+// GetEventPoliciesForResource returns the applying EventPolicies for a given resource
+func GetEventPoliciesForResource(lister listerseventingv1alpha1.EventPolicyLister, resourceGVK schema.GroupVersionKind, resourceObjectMeta metav1.ObjectMeta) ([]*v1alpha1.EventPolicy, error) {
+	policies, err := lister.EventPolicies(resourceObjectMeta.GetNamespace()).List(labels.Everything())
+	if err != nil {
+		return nil, fmt.Errorf("failed to list eventpolicies: %w", err)
+	}
+
+	relevantPolicies := []*v1alpha1.EventPolicy{}
+	resourceAPIVersion := fmt.Sprintf("%s/%s", resourceGVK.Group, resourceGVK.Version)
+
+	for _, policy := range policies {
+		for _, to := range policy.Spec.To {
+
+			if to.Ref != nil &&
+				strings.EqualFold(to.Ref.Name, resourceObjectMeta.GetName()) &&
+				strings.EqualFold(to.Ref.APIVersion, resourceAPIVersion) &&
+				strings.EqualFold(to.Ref.Kind, resourceGVK.Kind) {
+
+				relevantPolicies = append(relevantPolicies, policy)
+				break // no need to check the other .spec.to's from this policy
+			}
+
+			if to.Selector != nil &&
+				strings.EqualFold(to.Selector.APIVersion, resourceAPIVersion) &&
+				strings.EqualFold(to.Selector.Kind, resourceGVK.Kind) {
```

Comment:
Do we want to match the full APIVersion and Kind or just kind + group ? what happens when there is a new version?

Reply:
I think we thought about that initially (https://docs.google.com/document/d/1u6vyMviGpD13uyLKSGV7_D_p0ZRBPWg6PlLItRH8UFI/edit?resourcekey=0-gCE8brkIopDqc4jE7c9_tQ&disco=AAABK_fB-1s). The problem there was somehow related, that the client needs to know the version. But as we don't need that connection (e.g. from EP reconciler to a broker resource), I think this requirement is not needed IIRC... :thinking: (or maybe I am missing something ATM...)

But then we could also change the API in the `.to.ref` from `apiVersion` to `group` :thinking:

Reply:
Updated as discussed

Reply:
Damn. I deleted my original comment by accident :facepalm: 
Only for history: We thought about this [initially] (https://docs.google.com/document/d/1u6vyMviGpD13uyLKSGV7_D_p0ZRBPWg6PlLItRH8UFI/edit?resourcekey=0-gCE8brkIopDqc4jE7c9_tQ&disco=AAABK_fB-1s) in the design doc , but then discarded it, because we needed the full api version for the client. I wasn't sure then, if this was still a requirement, but it turned out, we need it for the client/tracker.